perception/ground_segmentation/src/ray_ground_filter_nodelet.cpp

```diff
@@ -257,20 +257,59 @@ void RayGroundFilterComponent::ClassifyPointCloud(
 //   return (true);
 // }
 
-void RayGroundFilterComponent::ExtractPointsIndices(
-  const pcl::PointCloud<PointType_>::Ptr in_cloud_ptr, const pcl::PointIndices & in_indices,
-  pcl::PointCloud<PointType_>::Ptr out_only_indices_cloud_ptr,
-  pcl::PointCloud<PointType_>::Ptr out_removed_indices_cloud_ptr)
+void RayGroundFilterComponent::initializePointCloud2(
+  const PointCloud2::ConstSharedPtr & in_cloud_ptr, PointCloud2::SharedPtr & out_cloud_msg_ptr)
 {
-  pcl::ExtractIndices<PointType_> extract_ground;
-  extract_ground.setInputCloud(in_cloud_ptr);
-  extract_ground.setIndices(pcl::make_shared<pcl::PointIndices>(in_indices));
+  out_cloud_msg_ptr->header = in_cloud_ptr->header;
+  out_cloud_msg_ptr->height = in_cloud_ptr->height;
+  out_cloud_msg_ptr->fields = in_cloud_ptr->fields;
+  out_cloud_msg_ptr->is_bigendian = in_cloud_ptr->is_bigendian;
+  out_cloud_msg_ptr->point_step = in_cloud_ptr->point_step;
+  out_cloud_msg_ptr->is_dense = in_cloud_ptr->is_dense;
+  out_cloud_msg_ptr->data.resize(in_cloud_ptr->data.size());
+}
 
-  extract_ground.setNegative(false);  // true removes the indices, false leaves only the indices
-  extract_ground.filter(*out_only_indices_cloud_ptr);
+void RayGroundFilterComponent::ExtractPointsIndices(
```

Comment:
I guess following logic will more faster and easier to understand. Because memcpy is fast enough and sort will cost O(n logn).

```c++
// get pointcloud indices: O(n)
std::vector<bool> is_ground_point(in_cloud_ptr->width, false);
for (int idx : in_indices.indices) {
    is_ground_point[idx] = true;
}

// copy: O(n)
for (size_t i = 0; i < is_ground_point.size(); ++i) {
    if (is_ground_point[i]) {
        // copy ground point
    } else {
        // copy no ground point
    }
}
```

Reply:
@YoshiRi Thank you for your comment. 
I changed as your recommendation at https://github.com/autowarefoundation/autoware.universe/pull/6791/commits/5cf469e770a838d3cae325ecd82e879961cdef70

Reply:
Thanks. LGTM